Fix getting drop target when dragging from file manager

PVince81 opened the pull request:
## Description
Possibly related to jquery.fileupload upgrade, now
e.originalEvent.target might be null.

## Related Issue
Fixes https://github.com/owncloud/core/issues/28689.

## Motivation and Context
See ticket

## How Has This Been Tested?
Manual testing like in the ticket.

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

Commit:
Fix getting drop target when dragging from file manager

Related to jquery.fileupload upgrade, the event has moved

jvillafanez commented on apps/files/js/filelist.js:
```diff
@@ -2708,7 +2708,7 @@
 				}
 
-				var dropTarget = $(e.originalEvent.target);
+				var dropTarget = $(e.originalEvent.target || e.originalEvent.delegatedEvent.target);
```
I guess the `e.original.target` won't be needed any longer.

The change in the library that affects this is https://github.com/owncloud/core/blob/639dafdd4db2eb72210b623b1a66c427b42b14d4/apps/files/js/jquery.fileupload.js#L1137 (before the library update) and https://github.com/owncloud/core/blob/master/apps/files/js/jquery.fileupload.js after the library update.
The actual event isn't set as the originalEvent but as the delegateEvent.

Unless we want compatibility between the library versions, which I don't think it's needed, we can remove the old code.

PVince81 replied:
Thanks for looking deeper. To be honest I was just too lazy to investigate this detail.

PVince81 replied:
@jvillafanez adjusted

SergioBertolinSG commented:
When uploading several files to a folder there is no uploading bar.

SergioBertolinSG commented:
When uploading a folder with several folders and files inside them, the files are not uploaded.

PVince81 commented:
stable10: https://github.com/owncloud/notifications/pull/117

jvillafanez commented:
Wrong link? Backport doesn't seem related.

PVince81 commented:
oh lala, major browser tab fail. I'll fix

jvillafanez commented:
> When uploading a folder with several folders and files inside them, the files are not uploaded.

Folder upload seems to be be supported only in chrome. It's very likely a browser limitation.

SergioBertolinSG commented:
Using chrome I mean.

PVince81 commented:
- [ ] BUG: progress bar does not disappear when uploading a folder (chrome) into the current dir

PVince81 commented:
- [ ] BUG: target row size must be refreshed after transfer to have the correct size

I had a look at this and it's more complex, to be fixed separately. This is not related to the jquery.fileupload library update not recent changes. It is likely present since we moved to Webdav for file operations.

Commit:
Always pass requesttoken on uploads

Commit:
Progress bar must disappear at the of upload

When uploading multiple files into a target folder, sometimes the
progress bar does not disappear.

PVince81 commented:
@SergioBertolinSG please retest

PVince81 commented:
- [ ] fix js tests

PVince81 commented:
tests fixed

SergioBertolinSG commented:
The estimation of the upload bar is very general (few seconds, minutes, etc) has it changed in this PR?

SergioBertolinSG commented:
I guess it didn't so it works fine.

SergioBertolinSG reviewed (approved):
Works fine.

PVince81 commented:
@SergioBertolinSG the estimate was never good...

jvillafanez reviewed (approved):
Code looks good

PVince81 commented:
stable10: https://github.com/owncloud/core/pull/28882
